### PR TITLE
"Authenticate with OAuth" sample

### PR DESCRIPTION
**Pull request by PaulAllanSturm**

### CODE REVIEW

File: lib/src/samples/authenticate_with_oauth.dart [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import 'package:flutter/material.dart';
```
Comment (jenmerritt) [inline lib/src/samples/authenticate_with_oauth.dart:1]: Need to add the apache licenses for both sample files

File: lib/src/samples/authenticate_with_oauth_sample.dart [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      ),
+    );
+  }
+
+  @override
+  void dispose() async {
+    super.dispose();
+
+    ArcGISEnvironment
+        .authenticationManager.arcGISAuthenticationChallengeHandler = null;
```
Comment (jenmerritt) [inline lib/src/samples/authenticate_with_oauth_sample.dart:52]: Maybe a comment for this step too? Just to emphasize it as part of the workflow.

**Review by jenmerritt — APPROVED** [review]
Looks good! Just needs the copyright statements, and one other optional doc suggestion.

File: lib/src/samples/authenticate_with_oauth.dart (hunk 1 above)
Comment (PaulAllanSturm) [reply lib/src/samples/authenticate_with_oauth.dart:17]: added

File: lib/src/samples/authenticate_with_oauth_sample.dart (hunk 2 above)
Comment (PaulAllanSturm) [reply lib/src/samples/authenticate_with_oauth_sample.dart:70]: added

**Review by jenmerritt — APPROVED** [review]
👍

**Pull request merged**